MathLib/LinAlg/Lis/LisOption.h

```diff
@@ -17,88 +17,43 @@
 
 #include <string>
 #include <map>
+#include <boost/algorithm/string.hpp>
+
+#include <logog/include/logog.hpp>
 
 #include "BaseLib/ConfigTree.h"
 
 namespace MathLib
 {
-
-/**
- * \brief Option for Lis solver
+/** Option for Lis solver.
+ *
+ * Options are stored as a strings obtained from lis-tag of the passed options.
+ * The string will be passed as is to the lis.  In particular it will not be
+ * checked by this method if options are valid.
+ *
+ * For possible values, please refer to the Lis User Guide:
+ * http://www.ssisc.org/lis/lis-ug-en.pdf
+ *
+ * Note: Option -omp_num_threads cannot be used with this class since Lis
+ * currently (version 1.5.57) only sets the number of threads in
+ * \c lis_initialize(). Refer to the Lis source code for details.
  */
 struct LisOption
 {
-    using Key = std::string;
-    using Value = std::string;
+	LisOption(BaseLib::ConfigTree const* const options)
+	{
+		if (options)
+		{
+			_option_string += " " + options->get<std::string>("lis", "");
+			boost::algorithm::trim_right(_option_string);
```

Comment:
This would need an extra space and a trim anyway:
```c++
auto const s = options->get_optional<std::string>("lis");
if (s) _option_string += " " + *s;
trim_right(_option_string);
```